common/solver/batch_bicgstab_kernels.hpp.inc

```diff
@@ -0,0 +1,847 @@
+
+
+        const auto A_global_entry =
+            gko::batch::batch_entry(gko::batch::to_const(a), ibatch);
+
+        const gko::batch_dense::BatchEntry<const ValueType> b_global_entry =
+            gko::batch::batch_entry(gko::batch::to_const(b), ibatch);
+
+        const gko::batch_dense::BatchEntry<ValueType> x_global_entry =
+            gko::batch::batch_entry(x, ibatch);
+
+
+        const gko::batch_dense::BatchEntry<ValueType> x_shared_entry{
+            x_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> r_shared_entry{
+            r_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> r_hat_shared_entry{
+            r_hat_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> p_shared_entry{
+            p_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> p_hat_shared_entry{
+            p_hat_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> v_shared_entry{
+            v_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> s_shared_entry{
+            s_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> s_hat_shared_entry{
+            s_hat_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> t_shared_entry{
+            t_sh, static_cast<size_type>(nrhs), nrows, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> rho_old_shared_entry{
+            rho_old_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> rho_new_shared_entry{
+            rho_new_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> omega_old_shared_entry{
+            omega_old_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> omega_new_shared_entry{
+            omega_new_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> alpha_shared_entry{
+            alpha_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<ValueType> beta_shared_entry{
+            beta_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<real_type> rhs_norms_shared_entry{
+            norms_rhs_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<real_type> res_norms_shared_entry{
+            norms_res_sh, static_cast<size_type>(nrhs), 1, nrhs};
+
+        const gko::batch_dense::BatchEntry<real_type>
+            res_norms_temp_shared_entry{norms_res_temp_sh,
+                                        static_cast<size_type>(nrhs), 1, nrhs};
```

Comment:
Can we abstract these initializations away in a function ?

Reply:
I guess that is not possible as discussed.